website/docs/HowShouldIQuery.md

```diff
@@ -3,19 +3,143 @@ id: how-should-i-query
+
+Here are general guidelines for picking idiomatic query variants:
+
+1. Use `getBy*` in the most common case when you expect a **single matching element**. Use other queries only in more specific cases.
+2. Use `findBy*` when an element is not yet in the element tree, but you expect it to be there as a **result of some asynchronous action**.
+3. Use `getAllBy*` (and `findAllBy*` for async) if you expect **more than one matching element**, e.g. in a list.
+4. Use `queryBy*` variant only when element **should not exist** to use it together with e.g. [`not.toBeOnTheScreen()`](jest-matchers#tobeonthescreen) matcher.
+
+Do not use `queryAllBy*` as it provides no assertions on the number of matching elements.
+
+## Query predicate
+
+The query predicate describes how you decide whether to match the given element.
+
+| Predicate                                               | Supported elements | Inspected props                                                                             |
+| ------------------------------------------------------- | ------------------ | ------------------------------------------------------------------------------------------- |
+| [`*ByRole`](api-queries#by-role)                        | all host elements  | `role`, `accessibilityRole`,<br /> optional: accessible name, accessibility state and value |
+| [`*ByLabelText`](api-queries#by-label-text)             | all host elements  | `aria-label`, `aria-labelledby`,<br /> `accessibilityLabel`, `accessibilityLabelledBy`      |
+| [`*ByDisplayValue`](api-queries#by-display-value)       | `TextInput`        | `value`, `defaultValue`                                                                     |
+| [`*ByPlaceholderText`](api-queries#by-placeholder-text) | `TextInput`        | `placeholder`                                                                               |
+| [`*ByText`](api-queries#by-text)                        | `Text`             | `children` (text content)                                                                   |
+| [`*ByHintText`](api-queries#by-hint-text)               | all host elements  | `accessibilityHint`                                                                         |
+| [`*ByTestId`](api-queries#by-test-id)                   | all host elements  | `testID`                                                                                    |
+
+### Idiomatic query predicates
+
+Choosing the proper query predicate helps better express the test's intent and make the tests resemble how users interact with your code (components, screens, etc.) as much as possible following our [Guiding Principles](https://testing-library.com/docs/guiding-principles).
+
+Additionally, most predicates promote the usage of proper accessibility props, which add a semantic layer on top of an element tree composed primarily of [`View`](https://reactnative.dev/docs/view) elements.
+
+### 1. By Role query {#by-role-query}
```

Comment:
i'm curious, what's the use of `{#by-role-query}`? it's the first time i see this

Reply:
This is definition of anchor for http links in Docosaurus